### PR TITLE
[Hub Generated] Review request for Creator to add version preview/2022-09-01-preview

### LINKED ISSUE #19690: [Maps - Wayfinder] API Review
Service Name: Azure Maps Wayfinder Service

Key contact for the review: Xubin Zhuge(@xubin Zhuge) / James Jonghan Park(@james Jonghan Park)

### PR DESCRIPTION
**Pull request by davsta90**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/creatorZN...main/).
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*engagement record*](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/271/Azure-REST-API-Stewardship?anchor=rest-api-stewardship-process).

* Link to engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [x] a public preview
- [ ] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous Open API document (swagger) if applicable, and the root paths that have been updated. </sup>
* Design Document:
* Previous Open API Doc:
* Updated paths:

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azapi) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azapi).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

**Comment by davsta90** [comment]
there are model validation fixes which seem to be false positives:

ROUTESET.JSON
ERROR (DeleteOperationResponses/R4011/ARMViolation): The delete operation is defined without a 200 or 204 error response implementation,please add it.'

WAYFIND.JSON and ROUTESET.JSON:
ERROR (OperationsAPIImplementation/R3023/ARMViolation): Operations API must be implemented for '/providers//operations'.
ERROR (SecurityDefinitionsStructure/R2054/SDKViolation): Every swagger/configuration must have a security definitions section and it must adhere to the structure described in: https://github.com/Azure/azure-openapi-validator/blob/master/docs/security-definitions-structure-validation.md

File: specification/maps/data-plane/Creator/preview/2022-09-01-preview/wayfind.json [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+            "NoExplanation"
+          ],
+          "x-ms_enum": {
+            "name": "noResultExplanation",
+            "description": "In the event that no paths are found, the `paths` array will be empty. If the reason no paths were found can be determined, this property will provide an explanation why.",
+            "values": [
+              "NoExplanation"
+            ],
+            "modelAsString": true
+          }
```
Comment (weidongxu-microsoft) [inline specification/maps/data-plane/Creator/preview/2022-09-01-preview/wayfind.json:147]: Given modelAsString=true, this could have more value than "NoExplanation". But do we have other code than this "NoExplanation" code to be documented?

File: specification/maps/data-plane/Creator/preview/2022-09-01-preview/wayfind.json [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+          "x-ms-identifiers": []
+        }
+      }
+    },
+    "Path": {
+      "description": "The details about a particular path between the origin and destination. It can be made up of one or more legs.",
+      "type": "object",
+      "properties": {
+        "lengthInMeters": {
+          "type": "number",
```
Comment (weidongxu-microsoft) [inline specification/maps/data-plane/Creator/preview/2022-09-01-preview/wayfind.json:112]: For "number", you might want to specify format=double or float.

**Review by weidongxu-microsoft — APPROVED** [review]

File: package-lock.json [hunk 3]
```diff
@@ -1446,9 +1446,9 @@
       }
     },
     "node_modules/node-fetch": {
```
Comment (weidongxu-microsoft) [inline package-lock.json:1448]: this file should not be included in the commit/PR

Comment (davsta90) [reply package-lock.json:1448]: I don't think it is my doing. I see the file in the main branch as well.

File: specification/maps/data-plane/Creator/preview/2022-09-01-preview/routeset.json [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+        "facilityId": {
+          "description": "the identifier of the structure this routeset was created from.",
+          "type": "string",
+          "readOnly": true
+        },
+        "levelOrdinals": {
+          "description": "the level identifiers within this facility",
+          "type": "array",
+          "items": {
+            "type": "number"
```
Comment (weidongxu-microsoft) [inline specification/maps/data-plane/Creator/preview/2022-09-01-preview/routeset.json:73]: It is integer or float? Ask because the description is "identifiers", and float seems not a good candidate.

File: specification/maps/data-plane/Creator/preview/2022-09-01-preview/routeset.json [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+      "in": "path",
+      "required": true,
+      "x-ms-parameter-location": "method"
+    }
+  },
+  "paths": {
+    "/routesets": {
+      "post": {
+        "description": "**Applies to:** see [pricing tiers](https://aka.ms/AzureMapsPricingTier).\n\nCreator makes it possible to develop applications based on your private indoor map data using Azure Maps API and SDK. See the [Creator for indoor maps](https://docs.microsoft.com/azure/azure-maps/creator-indoor-maps) article for an introduction to the Azure Maps Creator concepts and tools.\n\nThis API allows the caller to create routeset data from a dataset.\n\nUse the routeset API to create special data structures consumed by the Azure Maps Wayfinding service to discover the shortest paths within a facility.\n\n### Submit Create Request\n\nTo create a routeset, make a `POST` request with an empty body. The dataset is used as the source of the routeset data, so the `datasetId` query parameter must be included.\n\nThe Create Routeset API is a [long-running request](https://aka.ms/am-creator-lrt-v2).",
+        "operationId": "routesetCreate",
```
Comment (weidongxu-microsoft) [inline specification/maps/data-plane/Creator/preview/2022-09-01-preview/routeset.json:190]: For SDK, usually this would be "CreateRouteset". But if your team write the SDK, it does not matter.

File: specification/maps/data-plane/Creator/preview/2022-09-01-preview/routeset.json [hunk 6]
```diff
@@ -0,0 +1,360 @@
...
+      "description": "The ID used to query the status of the routeset creation.",
+      "type": "string",
+      "in": "path",
+      "required": true,
+      "x-ms-parameter-location": "method"
+    }
+  },
+  "paths": {
+    "/routesets": {
+      "post": {
```
Comment (weidongxu-microsoft) [inline specification/maps/data-plane/Creator/preview/2022-09-01-preview/routeset.json:188]: I think the general recommendation is a PUT. But maybe we already discussed this in meeting, so ignore me if this already settled.

File: specification/maps/data-plane/Creator/preview/2022-09-01-preview/routeset.json [hunk 7]
```diff
@@ -0,0 +1,360 @@
...
+        "parameters": [
+          {
+            "$ref": "../../../Common/preview/1.0/common.json#/parameters/ClientId"
+          },
+          {
+            "$ref": "#/parameters/ApiVersion"
+          }
+        ],
+        "x-ms-pageable": {
+          "itemName": "tilesets",
```
Comment (weidongxu-microsoft) [inline specification/maps/data-plane/Creator/preview/2022-09-01-preview/routeset.json:240]: There is no "tilesets" in "RoutesetsList".

Is it "routesets"?

File: specification/maps/data-plane/Creator/preview/2022-09-01-preview/routeset.json [hunk 8]
```diff
@@ -0,0 +1,360 @@
...
+          "200": {
+            "description": "The operation is running or complete. If the operation was successful, use the Resource-Location header to obtain the path to the result.",
+            "schema": {
+              "$ref": "./tileset.json#/definitions/LongRunningOperationResult"
+            },
+            "headers": {
+              "Resource-Location": {
+                "type": "string",
+                "description": "If successful, a URI where details on the newly created resource can be found."
+              }
```
Comment (weidongxu-microsoft) [inline specification/maps/data-plane/Creator/preview/2022-09-01-preview/routeset.json:350]: This might not be a very standard LRO process. But OK if all map uses this pattern and your team write your own SDK.

Comment (davsta90) [reply specification/maps/data-plane/Creator/preview/2022-09-01-preview/routeset.json:350]: this is how the other creator services API's do it. It was covered with the review board.

File: package-lock.json (hunk 3 above)
Comment (davsta90) [reply package-lock.json]: I updated the file to match the version in the master branch

File: specification/maps/data-plane/Creator/preview/2022-09-01-preview/wayfind.json (hunk 1 above)
Comment (davsta90) [reply specification/maps/data-plane/Creator/preview/2022-09-01-preview/wayfind.json:147]: initially it was going to be just a string, but the review board suggested using an extensible enum.  for now it can only have a value and it will expand in the future as we learn more about the behavior.

File: specification/maps/data-plane/Creator/preview/2022-09-01-preview/routeset.json (hunk 4 above)
Comment (davsta90) [reply specification/maps/data-plane/Creator/preview/2022-09-01-preview/routeset.json:73]: it is an integer identifying the floor.it value is required in the wayfind API to identify which floor the endpoints refer to. that said, thanks for bringing this up: I reworded some of the descriptions to improve clarity.

File: specification/maps/data-plane/Creator/preview/2022-09-01-preview/routeset.json (hunk 5 above)
Comment (davsta90) [reply specification/maps/data-plane/Creator/preview/2022-09-01-preview/routeset.json:190]: updated to "Routeset_Create" to model what done in our SDK

File: specification/maps/data-plane/Creator/preview/2022-09-01-preview/wayfind.json (hunk 2 above)
Comment (davsta90) [reply specification/maps/data-plane/Creator/preview/2022-09-01-preview/wayfind.json:112]: validation won't allow it. integer and number are the only numeric schema types allowed.

File: specification/maps/data-plane/Creator/preview/2022-09-01-preview/routeset.json (hunk 6 above)
Comment (davsta90) [reply specification/maps/data-plane/Creator/preview/2022-09-01-preview/routeset.json:188]: yes, we already GA services doing POST, so we do this to stay consistent.

File: specification/maps/data-plane/Creator/preview/2022-09-01-preview/routeset.json (hunk 7 above)
Comment (davsta90) [reply specification/maps/data-plane/Creator/preview/2022-09-01-preview/routeset.json:240]: copy/paste error: nice catch. thanks!

**Comment by weidongxu-microsoft** [comment]
Ignore BreakingChange and ModelValidation

BreakingChanges compare to 2022-09-01-preview in main and figure out several new APIs. Data-plane usually have different jsons.

ModelValidation on "LRO_RESPONSE_HEADER" only apply to ARM.
